src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -3908,6 +3908,13 @@ private bool IsConstructorAccessible(MethodSymbol constructor, ref HashSet<Diagn
             DiagnosticBag diagnostics,
             BoundExpression boundInitializerOpt = null)
         {
+
+            var typeContainingConstructors = type;
+            if (type.IsTupleType)
+            {
+                typeContainingConstructors = ((TupleTypeSymbol)type).UnderlyingTupleType;
```

Comment:
The intent is to search for constructors in the context of the underlying type. Everything else is unchanged - For example the type of the resulting expression node is the original type, not the underlying type.
I will add a comment.